---
title: "figures_annotations_of_bins"
author: "Amalia"
date: "2023-11-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#see how to annotate the bins you got
#We are going to annotate the top/bottom bins
We will UCSC annotation, and a waterfall concept, of the following order:
Promoters, Exons, Repetitive Elements, Introns, Rest of the genome.
Load packages
```{r}
library(GenomicRanges)
library(tidyverse)
#library (purrr)
library(plyranges)
#library(rlist)
#library(ggvenn)
#library(UpSetR)
library(viridis)
library(TxDb.Mmusculus.UCSC.mm9.knownGene)
```



```{r}
txdb<- TxDb.Mmusculus.UCSC.mm9.knownGene
#methods(class=class(txdb))
#I will keep the promoters of genes and not all transcripts
promoters<-promoters(genes(txdb, single.strand.genes.only=FALSE)) #keeps geens that have exons in both strands 
promoters<-unlist(promoters)
promoters<-disjoin_ranges(promoters)
introns<-intronsByTranscript(txdb) #theloun unlist
introns<-unlist(introns)
introns<-disjoin_ranges(introns)
exons<-exonsBy(txdb)
exons<-unlist(exons)
exons<-disjoin_ranges(exons)
exons
introns
repetitive<-readRDS("../data/repeat.instance.gr.rds") #tuncay gave me this one
disjoin_ranges(repetitive)

#1 promoters (they dont need to change cause they are the first)
promoter_ranges<-promoters
#2 exons (the common regions with promoters)
#setdiff removes the regions that are overlapping with a second GRanges obj
#i ignore strand cause in my genomic range si dont have strand info
#This might change in the next analysis
exon_ranges<-GenomicRanges::setdiff(exons, promoter_ranges, ignore.strand = TRUE)
#3 Repetitive elememts (the common regions with proms and exons must be removed)
repetitive_minus_proms<-GenomicRanges::setdiff(repetitive, promoter_ranges, ignore.strand= TRUE)
RE_ranges<-GenomicRanges::setdiff(repetitive_minus_proms, exon_ranges, ignore.strand =TRUE)
#4 introns (the common from proms, exons and repetitive must be removed)
introns_minus_prom<-GenomicRanges::setdiff(introns, promoter_ranges, ignore.strand = TRUE)
intr_min_prom_ex<-GenomicRanges::setdiff(introns_minus_prom, exon_ranges, ignore.strand = TRUE)
intron_ranges<-GenomicRanges::setdiff(intr_min_prom_ex, RE_ranges, ignore.strand = TRUE)
#5 everything that did not overlap will be REST
#just sanity check
sum(width(introns))
sum(width(introns_minus_prom))
sum(width(intr_min_prom_ex))
sum(width(intron_ranges))
```

bins with strand for genomic elements, bins without for chromhmm


NEW IDEA PLOT WITH THE ANNOTATION OF EVERY BIN AS BACKROUND
DO IT FIRST FOR THE STRANDED BINS
```{r}
sample_names<-c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B") #modified
addback<-c("DNMT3A2", "DNMT3A2", "DNMT3A1", "DNMT3B1","DNMT3B1","DNMT3B1","WT","WT")
tobind<-data.frame() #used to rbind all dfs together
for(i in 1:8){
a<-readRDS(file = paste("../data/processed/bins_with_strand_information/CpGs/1kb/top1%/", sample_names[i],".rds",sep=""))
a<-a %>% makeGRangesFromDataFrame(keep.extra.columns = T)  #convert to granges
##PROMOTERS 1 

#1 overlaps
hits<-findOverlaps(a,promoter_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],promoter_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_promoter_bina<-data.frame("row.number"=queryHits(hits),"promoter_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_promoter_bina<-length_promoter_bina %>% group_by(row.number) %>% summarise(across(promoter_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_promoter_bina<-length_promoter_bina %>% mutate(promoter_overlap=replace(promoter_overlap, promoter_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_promoter_bina<-left_join(to_merge,length_promoter_bina, by="row.number")
#sanity check
check_full_length_promoter_bina<-full_join(to_merge,length_promoter_bina, by="row.number")

##EXONS 2
#1 overlaps
hits<-findOverlaps(a,exon_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],exon_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_exons_bina<-data.frame("row.number"=queryHits(hits),"exon_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_exons_bina<-length_exons_bina %>% group_by(row.number) %>% summarise(across(exon_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_exons_bina<-length_exons_bina %>% mutate(exon_overlap=replace(exon_overlap, exon_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_exons_bina<-left_join(to_merge,length_exons_bina, by="row.number")

##REPETITIVE 3
#1 overlaps
hits<-findOverlaps(a,RE_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],RE_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_re_bina<-data.frame("row.number"=queryHits(hits),"re_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_re_bina<-length_re_bina %>% group_by(row.number) %>% summarise(across(re_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_re_bina<-length_re_bina %>% mutate(re_overlap=replace(re_overlap, re_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_re_bina<-left_join(to_merge,length_re_bina, by="row.number")

##INTRONS 4
#1 overlaps
hits<-findOverlaps(a,intron_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],intron_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_introns_bina<-data.frame("row.number"=queryHits(hits),"intron_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_introns_bina<-length_introns_bina %>% group_by(row.number) %>% summarise(across(intron_overlap, sum))
#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_introns_bina<-length_introns_bina %>% mutate(intron_overlap=replace(intron_overlap, intron_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_introns_bina<-left_join(to_merge,length_introns_bina, by="row.number")

#now merge everything and what is left is rest of genome
bina_overlaps<-left_join(full_length_promoter_bina,full_length_exons_bina) %>% left_join(.,full_length_re_bina) %>% left_join(.,full_length_introns_bina) %>% rowwise() %>% mutate(sum_overlap= sum(c(promoter_overlap ,exon_overlap , re_overlap,intron_overlap), na.rm=TRUE )) %>% mutate(Rest_overlap= 1000-sum_overlap) %>%dplyr::select(-c(sum_overlap, row.number)) %>% replace(is.na(.), 0)
#bina_overlaps


#pivot to longer for plot
bina_long<-bina_overlaps %>% pivot_longer(!bin, names_to = "region", values_to = "range") %>% mutate("sample_name" = print(sample_names[i]))

tobind<-rbind(tobind,bina_long)
#how to merge the two
#plot for two
#bina_long %>% ggplot(aes(y=region, fill=range)) +geom_bar(position= "fill")
}
 
```
plot individuallyn like this :

```{r}
tobind2<-tobind
tobind2$sample_name<- factor(tobind2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B"))
tobind2 %>% filter(sample_name %in%c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748")) %>% filter(region=="exon_overlap") %>% ggplot(aes(x=sample_name, y=range, fill=sample_name))+geom_bar(stat = "summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1)) + scale_fill_viridis_d()



plot_in<-tobind2 %>% filter(sample_name %in%c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748")) %>% filter(region=="exon_overlap") %>% ggplot(aes(x=sample_name, y=range, fill=sample_name))+geom_bar(stat = "summary") + labs(x="", y="", fill="", title = "Number of Promoters in top 1% ranked 1kb tiles") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/top1_percent.png", plot = plot)


```

```{r}
tobind2<-tobind
tobind2$sample_name<- factor(tobind2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B"))
tobind2 %>% filter(region=="exon_overlap") %>% ggplot(aes(x=sample_name, y=range, fill=sample_name))+geom_bar(stat = "summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1)) + scale_fill_viridis_d()
library(ggplot2)
library(ggdist)
library(tidyverse)
library(ggthemes)
library(tidyverse)
library(tidyquant)
library(ggdist)

tobind %>% 
  filter(region=="exon_overlap")%>% 
  ggplot(aes(x=sample_name, y= range, fill=sample_name)) +
  # add half-violin from {ggdist} package
  stat_halfeye(
    # adjust bandwidth
    adjust = 0.5,
    # move to the right
    justification = -0.2,
    # remove the slub interval
    .width = 0,
    point_colour = NA
  ) +
  geom_boxplot(
    width = 0.12,
    # removing outliers
    outlier.color = NA,
    alpha = 0.5
  ) +
  #stat_dots(
    # ploting on left side
  #  side = "left",
  #  position = position_jitter(width = 0.1),
    # adjusting position
  #  justification = 1.1,
  #  dotsize = 0.05,
    # adjust grouping (binning) of observations
  #  binwidth = 0.05,
  #  inherit.aes = TRUE,
  #  aes(fill = CGN_seq)
  #) 
  # Themes and Labels
  #scale_fill_manual(values = c("#F8766D", "#A3A500", "#00BF7D", "#00B0F6", "#E76BF3", "#FF7F00", "#FCBF1E", "#999999", "#E7CB94", "#56B4E9", "#009E73", "#0072B2", "#D55E00", "#CC79A7", "#F0E442", "#0072B2")) + # Set your desired colors here
  #scale_color_identity() +
  scale_fill_tq() +
  theme_tq()

plot<-tobind2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(position="identity",stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Top 1% ranked 1kb tiles") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +
  stat_summary(fun = "mean", geom = "text", aes(label = round(..y.., 2)),
               position = position_stack(vjust = 0.5)) 



#AUTO
plot<-tobind2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Top 1% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))+
  stat_summary(fun = "mean", geom = "text", aes(label = round(..y.., 0),colour="red"),
               position = position_stack(vjust = 0.5))

ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/with_numbers/top1_percent.png", plot = plot)

plot<-tobind_all2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Full  1kb tiles") + geom_text(data = overall_avg, aes(x = as.factor(1), y = range, label = round(range, 2)),
              position = position_dodge(width = 1), vjust = -0.5) +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))


```

```{r}
tobind
```

plot
```{r}
tobind %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

#convert sample name to factor to keep specific order
#test
tobind2<-tobind
tobind2$sample_name<- factor(tobind2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B"))

plot<-tobind2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Top 1% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/top1_percent.png", plot = plot)
  plot

#tobind2 %>% 
  
plot<-tobind2 %>% filter(!sample_name %in% c("GSM748786_as3A","GSM748786_as3B")) %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Top 1% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/top1_percent_noWT.png", plot = plot)
```

annotate all the bins
```{r}
sample_names<-c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786") #modified
addback<-c("DNMT3A2", "DNMT3A2", "DNMT3A1", "DNMT3B1","DNMT3B1","DNMT3B1","WT")
tobind_all<-data.frame() #used to rbind all dfs together
for(i in 1:7){
a<-readRDS(file = paste("../data/processed/bins_with_strand_information/CpGs/1kb/", sample_names[i],".rds",sep=""))
a<-a %>% makeGRangesFromDataFrame(keep.extra.columns = T)  #convert to granges
a<-a %>% filter(CG_num>4)
a<-a %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
a<- a %>% filter(Purine_num>1 & Pyrimidine_num>1)
##PROMOTERS 1 

#1 overlaps
hits<-findOverlaps(a,promoter_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],promoter_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_promoter_bina<-data.frame("row.number"=queryHits(hits),"promoter_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_promoter_bina<-length_promoter_bina %>% group_by(row.number) %>% summarise(across(promoter_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_promoter_bina<-length_promoter_bina %>% mutate(promoter_overlap=replace(promoter_overlap, promoter_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_promoter_bina<-left_join(to_merge,length_promoter_bina, by="row.number")
#sanity check
check_full_length_promoter_bina<-full_join(to_merge,length_promoter_bina, by="row.number")

##EXONS 2
#1 overlaps
hits<-findOverlaps(a,exon_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],exon_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_exons_bina<-data.frame("row.number"=queryHits(hits),"exon_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_exons_bina<-length_exons_bina %>% group_by(row.number) %>% summarise(across(exon_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_exons_bina<-length_exons_bina %>% mutate(exon_overlap=replace(exon_overlap, exon_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_exons_bina<-left_join(to_merge,length_exons_bina, by="row.number")

##REPETITIVE 3
#1 overlaps
hits<-findOverlaps(a,RE_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],RE_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_re_bina<-data.frame("row.number"=queryHits(hits),"re_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_re_bina<-length_re_bina %>% group_by(row.number) %>% summarise(across(re_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_re_bina<-length_re_bina %>% mutate(re_overlap=replace(re_overlap, re_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_re_bina<-left_join(to_merge,length_re_bina, by="row.number")

##INTRONS 4
#1 overlaps
hits<-findOverlaps(a,intron_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],intron_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_introns_bina<-data.frame("row.number"=queryHits(hits),"intron_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_introns_bina<-length_introns_bina %>% group_by(row.number) %>% summarise(across(intron_overlap, sum))
#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_introns_bina<-length_introns_bina %>% mutate(intron_overlap=replace(intron_overlap, intron_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_introns_bina<-left_join(to_merge,length_introns_bina, by="row.number")

#now merge everything and what is left is rest of genome
bina_overlaps<-left_join(full_length_promoter_bina,full_length_exons_bina) %>% left_join(.,full_length_re_bina) %>% left_join(.,full_length_introns_bina) %>% rowwise() %>% mutate(sum_overlap= sum(c(promoter_overlap ,exon_overlap , re_overlap,intron_overlap), na.rm=TRUE )) %>% mutate(Rest_overlap= 1000-sum_overlap) %>%dplyr::select(-c(sum_overlap, row.number)) %>% replace(is.na(.), 0)
#bina_overlaps


#pivot to longer for plot
bina_long<-bina_overlaps %>% pivot_longer(!bin, names_to = "region", values_to = "range") %>% mutate("sample_name" = print(sample_names[i]))

tobind_all<-rbind(tobind_all,bina_long)
#how to merge the two
#plot for two
#bina_long %>% ggplot(aes(y=region, fill=range)) +geom_bar(position= "fill")
saveRDS(tobind_all,file=paste0("../data/processed/bins_with_strand_information/CpGs/1kb/Annotated Bins_txdb/", "a_and_bfull.rds"))
}
```
plot
```{r}
tobind_all<-readRDS("../data/processed/bins_with_strand_information/CpGs/1kb/Annotated Bins_txdb/a_and_bfull.rds")
tobind_all %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

#convert sample name to factor to keep specific order
#test
tobind_all2<-tobind_all
tobind_all2$sample_name<- factor(tobind_all2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786"))

plot<-tobind_all2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of All 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))+
  stat_summary(fun = "mean", geom = "text", aes(label = round(..y.., 0),colour="red"),
               position = position_stack(vjust = 0.5))

ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/with_numbers/ALL.png.png", plot = plot)
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/ALL.png", plot = plot)
  plot
```

plot
```{r}
tobind_all<-readRDS("../data/processed/bins_with_strand_information/CpGs/1kb/Annotated Bins_txdb/a_and_bfull.rds")
tobind_all %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

#convert sample name to factor to keep specific order
#test
tobind2<-tobind_all
tobind2$sample_name<- factor(tobind2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B","GSM748787_as3A","GSM748787_as3B"))

plot<-tobind2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Full  1kb tiles") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d()
ggsave("../figures/genomic_Annotation_of_bins/1kb_windows/stranded_bins/ALL.png", plot = plot)
  plot
```
top 10 of stranded 



```{r}
sample_names<-c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B") #modified
addback<-c("DNMT3A2", "DNMT3A2", "DNMT3A1", "DNMT3B1","DNMT3B1","DNMT3B1","WT","WT")
tobind_top10<-data.frame() #used to rbind all dfs together
for(i in 1:8){
a<-readRDS(file = paste("../data/processed/bins_with_strand_information/CpGs/1kb/Top10%/", sample_names[i],".rds",sep=""))
a<-a %>% makeGRangesFromDataFrame(keep.extra.columns = T)  #convert to granges
##PROMOTERS 1 

#1 overlaps
hits<-findOverlaps(a,promoter_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],promoter_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_promoter_bina<-data.frame("row.number"=queryHits(hits),"promoter_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_promoter_bina<-length_promoter_bina %>% group_by(row.number) %>% summarise(across(promoter_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_promoter_bina<-length_promoter_bina %>% mutate(promoter_overlap=replace(promoter_overlap, promoter_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_promoter_bina<-left_join(to_merge,length_promoter_bina, by="row.number")
#sanity check
check_full_length_promoter_bina<-full_join(to_merge,length_promoter_bina, by="row.number")

##EXONS 2
#1 overlaps
hits<-findOverlaps(a,exon_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],exon_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_exons_bina<-data.frame("row.number"=queryHits(hits),"exon_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_exons_bina<-length_exons_bina %>% group_by(row.number) %>% summarise(across(exon_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_exons_bina<-length_exons_bina %>% mutate(exon_overlap=replace(exon_overlap, exon_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_exons_bina<-left_join(to_merge,length_exons_bina, by="row.number")

##REPETITIVE 3
#1 overlaps
hits<-findOverlaps(a,RE_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],RE_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_re_bina<-data.frame("row.number"=queryHits(hits),"re_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_re_bina<-length_re_bina %>% group_by(row.number) %>% summarise(across(re_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_re_bina<-length_re_bina %>% mutate(re_overlap=replace(re_overlap, re_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_re_bina<-left_join(to_merge,length_re_bina, by="row.number")

##INTRONS 4
#1 overlaps
hits<-findOverlaps(a,intron_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],intron_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_introns_bina<-data.frame("row.number"=queryHits(hits),"intron_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_introns_bina<-length_introns_bina %>% group_by(row.number) %>% summarise(across(intron_overlap, sum))
#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_introns_bina<-length_introns_bina %>% mutate(intron_overlap=replace(intron_overlap, intron_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_introns_bina<-left_join(to_merge,length_introns_bina, by="row.number")

#now merge everything and what is left is rest of genome
bina_overlaps<-left_join(full_length_promoter_bina,full_length_exons_bina) %>% left_join(.,full_length_re_bina) %>% left_join(.,full_length_introns_bina) %>% rowwise() %>% mutate(sum_overlap= sum(c(promoter_overlap ,exon_overlap , re_overlap,intron_overlap), na.rm=TRUE )) %>% mutate(Rest_overlap= 1000-sum_overlap) %>%dplyr::select(-c(sum_overlap, row.number)) %>% replace(is.na(.), 0)
#bina_overlaps


#pivot to longer for plot
bina_long<-bina_overlaps %>% pivot_longer(!bin, names_to = "region", values_to = "range") %>% mutate("sample_name" = print(sample_names[i]))

tobind_top10<-rbind(tobind_top10,bina_long)
#how to merge the two
#plot for two
#bina_long %>% ggplot(aes(y=region, fill=range)) +geom_bar(position= "fill")
}
 
```
plot
```{r}
tobind_top10 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

#convert sample name to factor to keep specific order
#test
tobind_top10_2<-tobind_top10
tobind_top10_2$sample_name<- factor(tobind_top10_2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B"))

plot<-tobind_top10_2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Top 10% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))+
  stat_summary(fun = "mean", geom = "text", aes(label = round(..y.., 0),colour="red"),
               position = position_stack(vjust = 0.5))

ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/with_numbers/top10_percent.png.png", plot = plot)
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/top10_percent.png", plot = plot)

  plot
```


bottom 1 of stranded 



```{r}
sample_names<-c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B") #modified
addback<-c("DNMT3A2", "DNMT3A2", "DNMT3A1", "DNMT3B1","DNMT3B1","DNMT3B1","WT","WT")
tobind_bottom1<-data.frame() #used to rbind all dfs together
for(i in 1:8){
a<-readRDS(file = paste("../data/processed/bins_with_strand_information/CpGs/1kb/Bottom1%/", sample_names[i],".rds",sep=""))
a<-a %>% makeGRangesFromDataFrame(keep.extra.columns = T)  #convert to granges
##PROMOTERS 1 

#1 overlaps
hits<-findOverlaps(a,promoter_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],promoter_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_promoter_bina<-data.frame("row.number"=queryHits(hits),"promoter_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_promoter_bina<-length_promoter_bina %>% group_by(row.number) %>% summarise(across(promoter_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_promoter_bina<-length_promoter_bina %>% mutate(promoter_overlap=replace(promoter_overlap, promoter_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_promoter_bina<-left_join(to_merge,length_promoter_bina, by="row.number")
#sanity check
check_full_length_promoter_bina<-full_join(to_merge,length_promoter_bina, by="row.number")

##EXONS 2
#1 overlaps
hits<-findOverlaps(a,exon_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],exon_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_exons_bina<-data.frame("row.number"=queryHits(hits),"exon_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_exons_bina<-length_exons_bina %>% group_by(row.number) %>% summarise(across(exon_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_exons_bina<-length_exons_bina %>% mutate(exon_overlap=replace(exon_overlap, exon_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_exons_bina<-left_join(to_merge,length_exons_bina, by="row.number")

##REPETITIVE 3
#1 overlaps
hits<-findOverlaps(a,RE_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],RE_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_re_bina<-data.frame("row.number"=queryHits(hits),"re_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_re_bina<-length_re_bina %>% group_by(row.number) %>% summarise(across(re_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_re_bina<-length_re_bina %>% mutate(re_overlap=replace(re_overlap, re_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_re_bina<-left_join(to_merge,length_re_bina, by="row.number")

##INTRONS 4
#1 overlaps
hits<-findOverlaps(a,intron_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],intron_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_introns_bina<-data.frame("row.number"=queryHits(hits),"intron_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_introns_bina<-length_introns_bina %>% group_by(row.number) %>% summarise(across(intron_overlap, sum))
#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_introns_bina<-length_introns_bina %>% mutate(intron_overlap=replace(intron_overlap, intron_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_introns_bina<-left_join(to_merge,length_introns_bina, by="row.number")

#now merge everything and what is left is rest of genome
bina_overlaps<-left_join(full_length_promoter_bina,full_length_exons_bina) %>% left_join(.,full_length_re_bina) %>% left_join(.,full_length_introns_bina) %>% rowwise() %>% mutate(sum_overlap= sum(c(promoter_overlap ,exon_overlap , re_overlap,intron_overlap), na.rm=TRUE )) %>% mutate(Rest_overlap= 1000-sum_overlap) %>%dplyr::select(-c(sum_overlap, row.number)) %>% replace(is.na(.), 0)
#bina_overlaps


#pivot to longer for plot
bina_long<-bina_overlaps %>% pivot_longer(!bin, names_to = "region", values_to = "range") %>% mutate("sample_name" = print(sample_names[i]))

tobind_bottom1<-rbind(tobind_bottom1,bina_long)
#how to merge the two
#plot for two
#bina_long %>% ggplot(aes(y=region, fill=range)) +geom_bar(position= "fill")
}
 
```

plot
```{r}
tobind_top10 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

#convert sample name to factor to keep specific order
#test
tobind_bottom1_2<-tobind_bottom1
tobind_bottom1_2$sample_name<- factor(tobind_bottom1_2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B"))

plot<-tobind_bottom1_2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Bottom 1% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d()+theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))+
  stat_summary(fun = "mean", geom = "text", aes(label = round(..y.., 0),colour="red"),
               position = position_stack(vjust = 0.5))

ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/with_numbers/bottom1_percent.png.png", plot = plot)
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/bottom1_percent.png", plot = plot)
  plot
  
plot<-tobind_bottom1_2 %>% filter(!sample_name %in% c("GSM748786_as3A","GSM748786_as3B")) %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Top 1% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/bottom1_percent_noWT.png", plot = plot)
```



bottom 10 of stranded 



```{r}
sample_names<-c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B") #modified
addback<-c("DNMT3A2", "DNMT3A2", "DNMT3A1", "DNMT3B1","DNMT3B1","DNMT3B1","WT","WT")
tobind_bottom10<-data.frame() #used to rbind all dfs together
for(i in 1:8){
a<-readRDS(file = paste("../data/processed/bins_with_strand_information/CpGs/1kb/Bottom10%/", sample_names[i],".rds",sep=""))
a<-a %>% makeGRangesFromDataFrame(keep.extra.columns = T)  #convert to granges
##PROMOTERS 1 

#1 overlaps
hits<-findOverlaps(a,promoter_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],promoter_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_promoter_bina<-data.frame("row.number"=queryHits(hits),"promoter_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_promoter_bina<-length_promoter_bina %>% group_by(row.number) %>% summarise(across(promoter_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_promoter_bina<-length_promoter_bina %>% mutate(promoter_overlap=replace(promoter_overlap, promoter_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_promoter_bina<-left_join(to_merge,length_promoter_bina, by="row.number")
#sanity check
check_full_length_promoter_bina<-full_join(to_merge,length_promoter_bina, by="row.number")

##EXONS 2
#1 overlaps
hits<-findOverlaps(a,exon_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],exon_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_exons_bina<-data.frame("row.number"=queryHits(hits),"exon_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_exons_bina<-length_exons_bina %>% group_by(row.number) %>% summarise(across(exon_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_exons_bina<-length_exons_bina %>% mutate(exon_overlap=replace(exon_overlap, exon_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_exons_bina<-left_join(to_merge,length_exons_bina, by="row.number")

##REPETITIVE 3
#1 overlaps
hits<-findOverlaps(a,RE_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],RE_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_re_bina<-data.frame("row.number"=queryHits(hits),"re_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_re_bina<-length_re_bina %>% group_by(row.number) %>% summarise(across(re_overlap, sum))

#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_re_bina<-length_re_bina %>% mutate(re_overlap=replace(re_overlap, re_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_re_bina<-left_join(to_merge,length_re_bina, by="row.number")

##INTRONS 4
#1 overlaps
hits<-findOverlaps(a,intron_ranges, ignore.strand=TRUE)
#2 width of overlap
b<-width(pintersect(a[queryHits(hits)],intron_ranges[subjectHits(hits)], ignore.strand=TRUE))
#3 create the df having id and length
length_introns_bina<-data.frame("row.number"=queryHits(hits),"intron_overlap"=b) 
#4 add the different overlaps for the same bin because many elements can overlap with the same bin
length_introns_bina<-length_introns_bina %>% group_by(row.number) %>% summarise(across(intron_overlap, sum))
#but we have to put a limit on the length of overlap (we can have a maximum of 1kb overlapping)
length_introns_bina<-length_introns_bina %>% mutate(intron_overlap=replace(intron_overlap, intron_overlap>1000, 1000))

#5 have to merge with original df but wont do it now cause there are too many columns and i want to see what im doing 

to_merge<-as.data.frame(a$bin)%>% dplyr::rename("bin" = "a$bin") %>% mutate(row.number=row_number())

full_length_introns_bina<-left_join(to_merge,length_introns_bina, by="row.number")

#now merge everything and what is left is rest of genome
bina_overlaps<-left_join(full_length_promoter_bina,full_length_exons_bina) %>% left_join(.,full_length_re_bina) %>% left_join(.,full_length_introns_bina) %>% rowwise() %>% mutate(sum_overlap= sum(c(promoter_overlap ,exon_overlap , re_overlap,intron_overlap), na.rm=TRUE )) %>% mutate(Rest_overlap= 1000-sum_overlap) %>%dplyr::select(-c(sum_overlap, row.number)) %>% replace(is.na(.), 0)
#bina_overlaps


#pivot to longer for plot
bina_long<-bina_overlaps %>% pivot_longer(!bin, names_to = "region", values_to = "range") %>% mutate("sample_name" = print(sample_names[i]))

tobind_bottom10<-rbind(tobind_bottom10,bina_long)
#how to merge the two
#plot for two
#bina_long %>% ggplot(aes(y=region, fill=range)) +geom_bar(position= "fill")
}
 
```


plot
```{r}
tobind_top10 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1))

#convert sample name to factor to keep specific order
#test
tobind_bottom10_2<-tobind_bottom10
tobind_bottom10_2$sample_name<- factor(tobind_bottom10_2$sample_name, levels=c("GSM4594635","GSM1382253","GSM2533056","GSM4837324","GSM1382256","GSM1545748","GSM748786_as3A","GSM748786_as3B"))

plot<-tobind_bottom10_2 %>% ggplot(aes(x=sample_name, y=range, fill=region)) +geom_bar(stat="summary") + labs(x="", y="", fill="", title = "Genomic Annotation of Bottom 10% ranked 1kb-sized genomic intervals") +theme_classic()+theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 7)) + scale_fill_viridis_d() +theme(axis.text.x=element_text(size=10), axis.text.y=element_text(size=10))+
  stat_summary(fun = "mean", geom = "text", aes(label = round(..y.., 0),colour="red"),
               position = position_stack(vjust = 0.5))

ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/with_numbers/bottom10_percent.png.png", plot = plot)
ggsave("../figures/Figures_for_report/genomic_annotation_of_bins/CpG/bottom10_percent.png", plot = plot)
  plot
```


chrom HMM


