---
title: "test_boxplot_of elements"
author: "Amalia"
date: "2023-11-10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#library(GenomicRanges)
library(tidyverse)
#library (purrr)
#library(plyranges)
#library(rlist)
library(ggvenn)
library(UpSetR)

library(viridis)
library(ggpointdensity)#for pointdensity


library(ggplot2)
library(ggridges)


library(ggplot2)
library(ggdist)
library(tidyverse)
library(ggthemes)
library(tidyverse)
library(tidyquant)
library(ggdist)
```

```{r}
sample_names<-c("GSM4594635","GSM1382253","GSM1382254","GSM2533056","GSM4837324","GSM1382255","GSM1382256","GSM1545748","GSM748786","GSM748787")
addback<-c("DNMT3A2", "DNMT3A2", "DNMT3A2", "DNMT3A1", "DNMT3B1","DNMT3B1","DNMT3B1","DNMT3B1","WT","WT") #I have to use these as names as well
```

long dataframe with all annotations for every sample, then boxplot of the score


```{r}
repetitive<-readRDS("../data/repeat.instance.gr.rds") #tuncay gave me this one

table(repetitive$E_class) #later
```

#distribution of genomic elements

##promoters CG 

```{r}
test_promoters_cg<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
a<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/promoters/", sample_names[i], ".RDS"))
b<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/exons_of_one_gene/", sample_names[i], ".RDS"))
e<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/introns_of_one_gene/", sample_names[i], ".RDS"))
f<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/RE/", sample_names[i], ".RDS"))
g<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/rest/", sample_names[i], ".RDS"))
a<-a %>% dplyr::select(c("gene_id","ratio_purpyr_log","CG_num","avg_meth")) %>% mutate(element="Promoter") %>% mutate(gene_id = as.character(gene_id))
b<-b %>% dplyr::select(c("gene_id","ratio_purpyr_log","CG_num","avg_meth")) %>% mutate(element="Exon") %>% mutate(gene_id = as.character(gene_id))
e<-e %>% dplyr::select(c("gene_id","ratio_purpyr_log","CG_num","avg_meth")) %>% mutate(element="Intron") %>% mutate(gene_id = as.character(gene_id))
f<-f %>% dplyr::select(c("names","ratio_purpyr_log","CG_num","avg_meth")) %>% mutate(element="RE") %>% rename(gene_id=names) %>% mutate(gene_id = as.character(gene_id))
g<-g %>% dplyr::select(c("row_number","ratio_purpyr_log","CG_num","avg_meth")) %>% mutate(element="Rest") %>% rename(gene_id=row_number) %>% mutate(gene_id = as.character(gene_id))
c<-bind_rows(a,b,e,f,g)

c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)


c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_promoters_cg<-rbind(test_promoters_cg,d)
}

```

```{r}
ggplot(c, aes(x = ratio_purpyr_log, y = element, height = stat(density))) +
  geom_density_ridges(stat = "density")





c %>% ggplot(aes(x=ratio_purpyr_log, color=element))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")

   plot<-c %>%
  ggplot(aes(x=element, y= ratio_purpyr_log, fill=element)) +
  # add half-violin from {ggdist} package
  stat_halfeye(
    # adjust bandwidth
    adjust = 0.5,
    # move to the right
    justification = -0.2,
    # remove the slub interval
    .width = 0,
    point_colour = NA
  ) +
  geom_boxplot(
    width = 0.12,
    # removing outliers
    outlier.color = NA,
    alpha = 1
  ) +
  stat_dots(
    # ploting on left side
    side = "left",
    position = position_jitter(width = 0.1),
    # adjusting position
    justification = 1.1,
    dotsize = 0.05,
    # adjust grouping (binning) of observations
    binwidth = 0.05,
    inherit.aes = TRUE,
    aes(fill = element)
  )+ 
     labs(x="", y="Methylation score", fill="", title = paste0(sample_names[i],"_",addback[i]))+
    theme_classic() +scale_fill_manual(values = c("#4e79a7","#a83232" ,"#6f5090","#ffa64d","black"))

```


```{r}
tail(test_promoters_cg)
```

```{r}
test_promoters_cg  %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_promoters_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_cg %>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_promoters_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_cg  %>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_promoters_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_promoters_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##promoters CG 

```{r}
test_rest<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/rest/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_rest<-rbind(test_rest,d)
}

```


```{r}
tail(test_rest)
```

```{r}
test_rest %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_rest %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_rest %>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_rest  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_rest %>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_rest  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_rest  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```


```{r}
test_rest %>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_rest  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##RE CG 

```{r}
test_RE_cg<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/RE/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_RE_cg<-rbind(test_RE_cg,d)
}

```


```{r}
tail(test_RE_cg)
```


```{r}
test_RE_cg %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_RE_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```


```{r}
test_RE_cg %>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_RE_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_cg %>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_RE_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_RE_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##promoters CG 

```{r}
test_introns_of_gene_cg<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/introns_of_one_gene/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_introns_of_gene_cg<-rbind(test_introns_of_gene_cg,d)
}

```


```{r}
tail(test_introns_of_gene_cg)
```

```{r}
test_introns_of_gene_cg  %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```


```{r}
test_introns_of_gene_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```


```{r}
test_introns_of_gene_cg  %>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_introns_of_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```


```{r}
test_introns_of_gene_cg  %>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```


```{r}
test_introns_of_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_of_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_of_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_introns_of_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##introns individual CG 

```{r}
test_introns_individual_cg<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/introns_individual/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_introns_individual_cg<-rbind(test_introns_individual_cg,d)
}

```


```{r}
tail(test_introns_individual_cg)
```


```{r}
test_introns_individual_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##exons of one gene CG 

```{r}
test_exons_of_one_gene_cg<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/exons_of_one_gene/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_exons_of_one_gene_cg<-rbind(test_exons_of_one_gene_cg,d)
}

```


```{r}
tail(test_exons_of_one_gene_cg)
```

```{r}
test_exons_of_one_gene_cg  %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```


```{r}
test_exons_of_one_gene_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_cg  %>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_exons_of_one_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_cg %>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.4) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")+scale_colour_manual(values = c("#4e79a7","#a83232" ,"#ffa64d"))+theme(axis.text.x=element_text(size=15, face = "bold"), axis.text.y=element_text(size=15, face = "bold"))
```

```{r}
test_exons_of_one_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_exons_of_one_gene_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##exons individual CG 

```{r}
test_exons_individual_cg<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/exons_individual/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_exons_individual_cg<-rbind(test_exons_individual_cg,d)
}

```


```{r}
tail(test_exons_individual_cg)
```


```{r}
test_exons_individual_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```
#distribution of genomic elements with chh data


##promoters CG 

```{r}
test_promoters_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/promoters/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_promoters_ch<-rbind(test_promoters_ch,d)
}

```


```{r}
tail(test_promoters_ch)
```


```{r}
test_promoters_ch %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_promoters_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_promoters_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##promoters CG 

```{r}
test_rest_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/rest/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_rest_ch<-rbind(test_rest_ch,d)
}

```


```{r}
tail(test_rest_ch)
```


```{r}
test_rest_ch %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_rest_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_rest_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_rest_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_rest_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_rest_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##RE Ch 

```{r}
test_RE_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/RE/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_RE_ch<-rbind(test_RE_ch,d)
}

```


```{r}
tail(test_RE_ch)
```


```{r}
test_RE_ch %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_RE_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_RE_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##promoters CG 

```{r}
test_introns_of_gene_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/introns_of_one_gene/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_introns_of_gene_ch<-rbind(test_introns_of_gene_ch,d)
}

```


```{r}
tail(test_introns_of_gene_ch)
```


```{r}
test_introns_of_gene_ch %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_of_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_of_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_of_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_of_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_introns_of_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##introns individual CG 

```{r}
test_introns_individual_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/introns_individual/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_introns_individual_ch<-bind_rows(test_introns_individual_ch,d)
}

```


```{r}
tail(test_introns_individual_cg)
```


```{r}
test_introns_individual_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_introns_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##exons of one gene CG 

```{r}
test_exons_of_one_gene_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/exons_of_one_gene/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_exons_of_one_gene_ch<-rbind(test_exons_of_one_gene_ch,d)
}

```


```{r}
tail(test_exons_of_one_gene_ch)
```


```{r}
test_exons_of_one_gene_ch %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_of_one_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_exons_of_one_gene_ch  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

##exons individual CG 

```{r}
test_exons_individual_ch<-data.frame() #empty dataframe to add to inside the loop
for(i in 1:8){
c<-readRDS(file=paste0("../data/processed/Annotation of genomic elements/CHH/exons_individual/", sample_names[i], ".RDS"))
c<-c %>% filter(CG_num>4)
c<-c %>% filter(avg_meth>0)
c<-c %>% filter(!is.na(Purine_num) & !is.na(Pyrimidine_num)) #have at least one from both
c<- c %>% filter(Purine_num>1 & Pyrimidine_num>1)
c<-c %>% mutate(Purine_meth_mod = Purine_meth + 0.01)
c<-c %>% mutate(Pyrimidine_meth_mod = Pyrimidine_meth + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr = Purine_meth_mod / Pyrimidine_meth_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_log = log2(ratio_pur_pyr)) 

#mutate to have ratio of number occurance
c<-c %>% mutate(Purine_num_mod = Purine_num + 0.01)
c<-c %>% mutate(Pyrimidine_num_mod = Pyrimidine_num + 0.01)
  #now create col with ratio of those two
c <-c %>% mutate(ratio_pur_pyr_num = Purine_num_mod / Pyrimidine_num_mod)
options(scipen = 999)# to avoid scientific notation
c<-c %>% mutate(ratio_purpyr_num_log = log2(ratio_pur_pyr_num)) 

d<-c %>% mutate(sample_name = sample_names[i])
d<-d %>% mutate(addback= addback[i])
test_exons_individual_ch<-rbind(test_exons_individual_ch,d)
}

```


```{r}
tail(test_exons_individual_cg)
```


```{r}
test_exons_individual_cg %>% filter(Purine_num>3 & Pyrimidine_num>3) %>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of methylation in CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3A1"|addback=="DNMT3A2") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3A", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="DNMT3B1") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "DNMT3B", x="log2 ratio of mapped CpHs followed by purine/pyrimidine") 
```

```{r}
test_exons_individual_cg  %>% filter(Purine_num>3 & Pyrimidine_num>3)%>% filter(addback=="WT") %>% ggplot(aes(x=ratio_purpyr_num_log, color=sample_name))+ geom_density() +theme_classic() + geom_vline(aes(xintercept=0),
            color="blue", linetype="dashed", linewidth = 0.1) +labs(title = "WT", x="log2 ratio of mapped CpHs followed by purine/pyrimidine")
```
